src/runtime_src/tools/xclbinutil/KernelUtilities.cxx

```diff
@@ -386,19 +408,44 @@ XclBinUtilities::addKernel(const boost::property_tree::ptree& ptKernel,
     // Create the new PS kernel instance and add it to the vector
     boost::property_tree::ptree ptIPEntry;
     ptIPEntry.put("m_type", "IP_PS_KERNEL");
+    // IP_PS_KERNEL specific data
+    // Note: extended-data is optional, only appliable to fixed ps kernel (-add-kernel)
+    //       not appicable to non-fixed ps kernel (-add-pskernel)
+    // "extended-data" {
+    //   "fundtional": "0"
+    //   "subtype": "1"
+    //   "dpu_kernel_id": "0x101"
+    // }   
+    boost::optional< const boost::property_tree::ptree& > boExtendedData = ptKernel.get_child_optional("extended-data");
+    if (boExtendedData) {
+      const boost::property_tree::ptree& ptExtendedData = boExtendedData.get();
+      if (!ptExtendedData.empty()) {
+        static const std::string NOT_DEFINED = "<not defined>";
+        auto subType = ptExtendedData.get<std::string>("subtype", NOT_DEFINED);
+        auto functional = ptExtendedData.get<std::string>("functional", NOT_DEFINED);
+        auto dpuKernelId = ptExtendedData.get<std::string>("dpu_kernel_id", NOT_DEFINED);
+        // subType and functinoal can either be string or numeric value
+        ptIPEntry.put("m_subtype", subType);
+        ptIPEntry.put("m_functional", functional);
+        ptIPEntry.put("m_kernel_id", dpuKernelId);
+      }
+    }
     ptIPEntry.put("m_base_address", "not_used");
     ptIPEntry.put("m_name", ipLayoutName);
-    ipLayout.push_back(ptIPEntry);
-    const unsigned int ipLayoutIndex = static_cast<unsigned int>(ipLayout.size()) - 1;
+
+    ipData.push_back(ptIPEntry);
+    const unsigned int ipLayoutIndex = static_cast<unsigned int>(ipData.size()) - 1;
 
     // -- For each PS Kernel Instance, connect any argument to its memory
     addArgsToMemoryConnections(ipLayoutIndex, ptKernel.get_child("arguments", ptEmpty), memTopology,  connectivity);
   }
 
   // Replace the original property tree
-  transformVectorToPtree(ipLayout, "ip_layout", "m_ip_data", ptIPLayoutRoot);
+  transformVectorToPtree(ipData, "ip_layout", "m_ip_data", ptIPLayoutRoot);
   transformVectorToPtree(connectivity, "connectivity", "m_connection", ptConnectivityRoot);
   transformVectorToPtree(memTopology, "mem_topology", "m_mem_data", ptMemTopologyRoot);
+
+  // XUtil::TRACE_PrintTree("IP_LAYOUT ROOT", ptIPLayoutRoot);
```

Comment:
Please remove deadcode if not being used.

Reply:
comment removed